File: 3 feb/lc78.py
from typing import *

class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        def generate(i,lst):
            if i == len(nums):
                ans.append(lst.copy())
                return
            generate(i+1,lst)
            lst.append(nums[i])
            generate(i+1,lst)
            lst.pop()
        generate(0,[])
        return ans
    
print(Solution().subsets([1,2]))

File: may/31may/lc78.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    void traverse(int[] nums,int i,List<Integer> arr){
        if(i==nums.length) return;
        arr.add(nums[i]);
        traverse(nums,i+1,arr);
        if(i == nums.length-1) ans.add(new ArrayList<>(arr));
        arr.remove(arr.size()-1);
        if(i == nums.length-1) ans.add(new ArrayList<>(arr));
        traverse(nums,i+1,arr);
    }
    public List<List<Integer>> subsets(int[] nums) {
        List<Integer> arr = new ArrayList<>();
        traverse(nums,0,arr);
        return ans;
    }
}